ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -596,14 +596,9 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
     final Optional<ColumnReferenceExp> fkColumnName =
         verifyJoin(root.getInfo(), preRepartitionLeft, preRepartitionRight);
 
-    final boolean finalJoin = prefix.isEmpty();
-
-    final JoinKey joinKey;
-    if (fkColumnName.isPresent()) {
-      joinKey = buildForeignJoinKey(root, fkColumnName.get());
-    } else {
-      joinKey = buildJoinKey(root);
-    }
+    final JoinKey joinKey = fkColumnName
+        .map(columnReferenceExp -> buildForeignJoinKey(root, fkColumnName.get()))
```

Comment:
Was not sure how to simplify (or even remove `buildForeignJoinKey()`...) cf https://github.com/confluentinc/ksql/pull/7522/files#r634680772